### ISSUE TITLE
Large convolution unit test in pyTorch gives seg fault

### ISSUE THREAD

**Issue by ashishfarmer** [ROCm/MIOpen#148]
A unit test in pyTorch that runs the config `convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1` gives a seg fault. The log from the test with MIOPEN_LOG_LEVEL=6 is attached here [log.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/4463478/log.txt)

This same config when I run through MIOpenDriver, there is a seg fault even before any convolution calls are executed.

<details><summary><b><i>Here is the log:</i></b></summary>

```
MIOPEN_LOG_LEVEL=6 /opt/rocm/miopen/bin/MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpenDriver: convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpen(HIP): Info [Handle] stream: 0x3b19fa0, device_id: 0
MIOpen(HIP): Info [ForwardGetWorkSpaceSize]
MIOpen(HIP): Info2 [HipGetHccVersionImpl] Running: '/opt/rocm-3.1.0/hcc/bin/hcc --version'
MIOpen(HIP): Info2 [HipGetHccVersionImpl] HCC clang version 10.0.0 (/data/jenkins-workspace/compute-rocm-rel-3.1/external/hcc-tot/llvm-project/clang 6a70953f87a209f37ea7884abdbb6bcb2d6db732) (based on HCC 3.1.20023-6d267cfb-6a70953f87a )
MIOpen(HIP): Info [HipGetHccVersionImpl] HCC base: 3.1.20023
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv2_COv3, MIOpen version 2.2.1.7633-rocm-rel-3.1-35-9218683
MIOpen(HIP): Info2 [ValidateGcnAssemblerImpl] Running: '/opt/rocm/hcc/bin/clang --version'
MIOpen(HIP): Info2 [ValidateGcnAssemblerImpl] HCC clang version 10.0.0 (/data/jenkins-workspace/compute-rocm-rel-3.1/external/hcc-tot/llvm-project/clang 6a70953f87a209f37ea7884abdbb6bcb2d6db732) (based on HCC 3.1.20023-6d267cfb-6a70953f87a )
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm3x3U: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm1x1U: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm1x1UV2: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm5x10u2v2f1: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm7x7c3h224w224k64u2v2p3q3f1: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvAsm5x10u2v2b1: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvOclDirectFwd11x11: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvOclDirectFwdGen: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvOclDirectFwd3x3: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvOclDirectFwd1x1: Not applicable
MIOpen(HIP): Info2 [GetWorkspaceSize] ConvOclDirectFwd: Not applicable
MIOpen(HIP): Info2 [SQLite_Db] Initializing system database file /opt/rocm-3.1.0/miopen/share/miopen/db/miopen.db
MIOpen(HIP): Info2 [SQLite_Db] Initializing user database file /root/.config/miopen//miopen.udb
MIOpen(HIP): Info2 [SQLite_Db] Database created successfully
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4R4Xdlops_1x1: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4R4FwdXdlops: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4_1x1: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4Fwd: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4R1Fwd: Not applicable
Segmentation fault (core dumped)
```
</details>
I am running on ROCm 3.1 docker on MI60 system, using the MIOpen 2.2 release that comes with ROCm 3.1.

---
EDIT: partial duplicate of #494

**Comment by daniellowell** [comment]
I will retest this on the newer stack.

**Comment by atamazov** [comment]
Input tensor requires 4 296 015 872 bytes of GPU memory (0x1 0010 0000). This means that byte offset in the input buffer exceeds 32 bits. IIRC our kernels use 32-bit offsets. Perhaps we need to add relevant checks to HIP, rocBLAS and OCL based Solvers. ASM-based Solvers already have necessary checks.

`FindConvFwdAlgorithm` begins with GEMM algo, so GEMM is the likely the first-order suspect.

**Comment by daniellowell** [comment]
@asroy Can you check if this is still an issue?

**Comment by asroy** [comment]
Still and issue, but it's not HIP iGEMM or ASM iGEMM, they are not applicable

MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
MIOpen(OpenCL): Info [Handle] Device name: gfx908
MIOpen(OpenCL): Info [Handle] stream: 0x281c510, device_id: 0x28cb170
MIOpen(OpenCL): Info [ForwardGetWorkSpaceSize] 
MIOpen(OpenCL): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, MIOpen version 2.10.0.8233-a361281a0
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] Running: '/opt/rocm/llvm/bin/clang --version'
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] clang version 11.0.0 (/src/external/llvm-project/clang ee4e4ebbadcc8ea14ce99e34ed31ab31e94827ac)
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] Target: x86_64-unknown-linux-gnu
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] Thread model: posix
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] InstalledDir: /opt/rocm/llvm/bin
MIOpen(OpenCL): Info2 [ValidateGcnAssemblerImpl] 
MIOpen(OpenCL): Info [GetFindModeValueImpl] MIOPEN_FIND_MODE = NORMAL(1)
MIOpen(OpenCL): Info2 [SQLiteBase] Initializing system database file /root/workspace/MIOpen/MIOpen-develop/src/kernels/miopen.db
MIOpen(OpenCL): Info2 [SQLiteBase] Initializing user database file /root/workspace/MIOpen/MIOpen-develop/src/kernels/miopen_1.0.0.udb
MIOpen(OpenCL): Info2 [HipCompilerVersionImpl] Read version information from HIP package...
MIOpen(OpenCL): Info [HipCompilerVersionImpl] 3.7.20315
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmForwardV4R5Xdlops: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmForwardV4R4Xdlops: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmBwdDataV4R1Xdlops: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmBwdDataV1R1Xdlops: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4R1Fwd: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmV4R4Fwd: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmBwdDataV1R1: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvHipImplicitGemmBwdDataV4R1: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvAsmImplicitGemmV4R1DynamicFwd_1x1: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvAsmImplicitGemmV4R1DynamicFwd: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvAsmImplicitGemmV4R1DynamicBwd: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvAsmImplicitGemmGTCDynamicFwdXdlops: Not applicable
MIOpen(OpenCL): Info2 [SearchForAllSolutions] ConvAsmImplicitGemmGTCDynamicBwdXdlops: Not applicable
MIOpen(OpenCL): Info2 [ForwardGetWorkSpaceSize] 0
/root//workspace/MIOpen/run.sh: line 255: 31224 Segmentation fault      (core dumped) bin/MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1

**Comment by fronteer** [comment]
1) The segment-fault occurs on in AllocateBuffersAndCopy() of driver/conv_driver.hpp, which is caused by the integer overflowing in GetTensorSize() of tensor_driver.hpp.  The code row

        size_t sz            = std::accumulate(len.begin(), len.end(), 1, std::multiplies<int>());

 using "int" rather than "size_t" for accumulating before the result is assigned to variable "sz" 

The following codes is able to remove this segment-fault 

            size_t sz = 1;

    for(const auto &elem : len)
        sz *= static_cast<size_t>(elem);

2)  Even after fixing the above issue, the testing case still fails and reports  "GPU Memory Access fault" with specific convolution kernel.  And the lots of following information are reported with "MIOPEN_LOG_LEVEL=6"

MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [CallGemm] gemm_desc: {isColMajor 0, transA 0, transB 0, m 2, n 4096, k 128, lda 128, ldb 4096, ldc 4096, batch_count 1, strideA 0, strideB 0, strideC 0, alpha 1, beta 0, dataType 0} 
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [CallGemm] gemm_desc: {isColMajor 0, transA 0, transB 0, m 2, n 4096, k 128, lda 128, ldb 4096, ldc 4096, batch_count 1, strideA 0, strideB 0, strideC 0, alpha 1, beta 0, dataType 0} 
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"

**Comment by fronteer** [comment]
More complete log: 

MIOpen(HIP): Info2 [SQLiteBase] Initializing user database file /root/.config/miopen/miopen_1.0.0.udb
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvBinWinograd3x3U: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvBinWinogradRxSf3x2: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvBinWinogradRxSf2x3: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvBinWinogradRxS: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd<3-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd<4-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd<5-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd<6-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd_xdlops<2-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd_xdlops<3-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd_xdlops<4-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd_xdlops<5-3>: Not applicable
MIOpen(HIP): Info2 [SearchForAllSolutions] ConvMPBidirectWinograd_xdlops<6-3>: Not applicable
MIOpen(HIP): Info2 [GetFindModeValueImpl] MIOPEN_FIND_MODE = HYBRID(3)
MIOpen(HIP): Info [GetForwardSolutions]
MIOpen(HIP): Info [Measure] Db::Prefetch time: 30.1978 ms
MIOpen(HIP): Info2 [FindRecordUnsafe] Looking for key 2-512-512-8x8-2-64-64-4097-0x0-8x8-1x1-0-NCHW-FP16-F in file /root/.config/miopen//gfx900_64.HIP.2_10_0_8236-04fd51a1.ufdb.txt
MIOpen(HIP): Info2 [FindRecordUnsafe] Key match: 2-512-512-8x8-2-64-64-4097-0x0-8x8-1x1-0-NCHW-FP16-F
MIOpen(HIP): Info2 [FindRecordUnsafe] Contents found: miopenConvolutionFwdAlgoGEMM:gemm,787.267,1048576,rocBlas,<unused>
MIOpen(HIP): Info2 [Measure] Db::FindRecord time: 0.097154 ms
MIOpen(HIP): Info2 [ForwardGetWorkSpaceSize] 1048576
MIOpen(HIP): Info [FindConvFwdAlgorithm] requestAlgoCount = 2, workspace = 1048576
MIOpen(HIP): Info [GetForwardSolutions]
MIOpen(HIP): Info2 [FindRecordUnsafe] Looking for key 2-512-512-8x8-2-64-64-4097-0x0-8x8-1x1-0-NCHW-FP16-F in file /root/.config/miopen//gfx900_64.HIP.2_10_0_8236-04fd51a1.ufdb.txt
MIOpen(HIP): Info2 [FindRecordUnsafe] Key match: 2-512-512-8x8-2-64-64-4097-0x0-8x8-1x1-0-NCHW-FP16-F
MIOpen(HIP): Info2 [FindRecordUnsafe] Contents found: miopenConvolutionFwdAlgoGEMM:gemm,787.267,1048576,rocBlas,<unused>
MIOpen(HIP): Info2 [Measure] Db::FindRecord time: 0.065641 ms
MIOpen(HIP): Info [CompileForwardSolution] solver_id = gemm
MIOpen(HIP): Info [FindConvFwdAlgorithm] miopenConvolutionFwdAlgoGEMM   787.267 1048576
MIOpen(HIP): Info [FindConvFwdAlgorithm] FW Chosen Algorithm: gemm , 1048576, 787.267
MIOpen(HIP): Info [ConvolutionForward] algo = 0, workspace = 1048576
MIOpen(HIP): Info2 [GetInvoker] Returning an invoker for problem 2x512x512x8x8x2x64x64x4097xNCHWxFP16x0x0x8x8x1x1x1xF and algorithm miopenConvolutionFwdAlgoGEMM
MIOpen(HIP): Info2 [GetFound1_0] No invokers found for 2x512x512x8x8x2x64x64x4097xNCHWxFP16x0x0x8x8x1x1x1xF
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernelDumpKernelParams] runcl MIOpenIm2d2Col.cl -k Im2d2Col -dumpilisa -r 10 if#0: if#0: if#0: iv#0 8192,1,1/256,1,1  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=2 -DNUM_IM_BLKS=16 -DLOCAL_MEM_SIZE=16384 -DSTRIDE_GT_1=1 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1
MIOpen(HIP): Info2 [SQLiteBase] Initializing system database file
MIOpen(HIP): Info [KernDb] database not present
MIOpen(HIP): Info2 [SQLiteBase] Initializing user database file /root/.cache/miopen/2.10.0.8236-04fd51a1/gfx900_64.ukdb
MIOpen(HIP): Info2 [KernDb] Database created successfully
MIOpen(HIP): Info2 [LoadBinary] Loading binary for: MIOpenIm2d2Col.cl ;args:  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=2 -DNUM_IM_BLKS=16 -DLOCAL_MEM_SIZE=16384 -DSTRIDE_GT_1=1 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -mcpu=gfx900
MIOpen(HIP): Info2 [Prepare] SELECT kernel_blob, kernel_hash, uncompressed_size FROM kern_db WHERE (kernel_name = 'MIOpenIm2d2Col.cl.o') AND (kernel_args = ' -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=2 -DNUM_IM_BLKS=16 -DLOCAL_MEM_SIZE=16384 -DSTRIDE_GT_1=1 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -mcpu=gfx900');
MIOpen(HIP): Info2 [Measure] Db::FindRecord time: 0.37696 ms
MIOpen(HIP): Info2 [LoadBinary] Sucessfully loaded binary for: MIOpenIm2d2Col.cl ;args:  -DNUM_CH_PER_WG=1 -DNUM_IM_BLKS_X=2 -DNUM_IM_BLKS=16 -DLOCAL_MEM_SIZE=16384 -DSTRIDE_GT_1=1 -DTILE_SZ_X=32 -DTILE_SZ_Y=8 -DUSE_IM_OFF_GUARD=1 -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_INT8x4=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -mcpu=gfx900
MIOpen(HIP): Info2 [CallGemm] gemm_desc: {isColMajor 0, transA 0, transB 0, m 2, n 4096, k 128, lda 128, ldb 4096, ldc 4096, batch_count 1, strideA 0, strideB 0, strideC 0, alpha 1, beta 0, dataType 0}
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [CallGemm] gemm_desc: {isColMajor 0, transA 0, transB 0, m 2, n 4096, k 128, lda 128, ldb 4096, ldc 4096, batch_count 1, strideA 0, strideB 0, strideC 0, alpha 1, beta 0, dataType 0}
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [CallGemm] gemm_desc: {isColMajor 0, transA 0, transB 0, m 2, n 4096, k 128, lda 128, ldb 4096, ldc 4096, batch_count 1, strideA 0, strideB 0, strideC 0, alpha 1, beta 0, dataType 0}
MIOpen(HIP): Info2 [GetKernels] 0 kernels for key: miopenIm2d2Col "c2i512_512w8_8p0_0s8_8d1_1t0"
MIOpen(HIP): Info2 [AddKernel] Key: miopenIm2Col "c2i512_512w8_8p0_0s8_8d1_1t0"

**Comment by asroy** [comment]
@fronteer Thanks for looking into it. Please try to narrow down the failure to the faulty kernel. You can disable solvers by setting env variables.
https://rocmsoftwareplatform.github.io/MIOpen/doc/html/DebugAndLogging.html#filtering-by-algorithm

**Comment by atamazov** [comment]
The fix for `size_t sz            = std::accumulate(len.begin(), len.end(), 1, std::multiplies<int>());` is available at #527

**Comment by atamazov** [comment]
[off-topic] @fronteer This is how to insert console logs:

````
<details><summary><b><i>HEADLINE</i></b></summary>

```
INSERT LOG CONTENTS HERE
```
</details>
````

**Comment by atamazov** [comment]
WRT GEMM issue I guess there is 4GiB buffer limit of rocBLAS. @shurale-nkn Am I right?

**Comment by atamazov** [comment]
Partial duplicate of #494

**Comment by asroy** [comment]
@atamazov So now host side is Ok, and the remaining issue is kernel cannot deal with >4GB buffer?

**Comment by atamazov** [comment]
@asroy WRT issue with GEMM algorithm, -- likely, yes.

**Comment by streamhsa** [comment]
@atamazov I'm not sure if there is an MIOpen kernel can support >4GB.

This config uses:
input: 4097*2*512*512*sizeof(half)= 4,296,015,872 Byte.

**Comment by asroy** [comment]
@atamazov I'm not sure if there is an MIOpen kernel can support >4GB.

This config uses:
input: 4097x2x512x512*sizeof(half)= 4,296,015,872 Byte.

**Comment by atamazov** [comment]
@asroy I think that both HIP and OCL kernels support 64-bit pointers. Consider the following config:
```
./bin/MIOpenDriver conv -W 256 -H 256 -c 256 -k 6 -x 1 -y 1 -n 128 -F 1
```
- Input tensor size (NCHW) = `128*256*256*256*4=` ___8 GiB___
- Output tensor size (NKHW) = `128*6*256*256*4=` 192 MiB
- Filter size (CKRS) = `256*6*1*1*4=` 6 kiB

Applicable Solvers (gfx908):
```
- id: 13 algo: 1, time: 11.1896 ms, ws: 0, name: ConvOclDirectFwd1x1
- id: 33 algo: 0, time: 24.3661 ms, ws: 0, name: gemm
- id: 37 algo: 3, time: 26.4706 ms, ws: 0, name: ConvBinWinogradRxSf3x2
```
All solutions pass verification (surprisingly, GEMM too!) @shurale-nkn Thoughts?

**Comment by ppanchad-amd** [comment]
Apologies for the lack of response. Please try latest ROCm 6.0.2 to see if your issue has been resolved? If resolved, please close the ticket. Thanks.

**Comment by atamazov** [comment]
The issue still persists Navi21, develop at 70f0225091a64bcd451428ec7107cf609be18426:

```bash
MIOPEN_LOG_LEVEL=4 ./bin/MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1 -i 1 -S 91
MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1 -i 1 -S 91
PRNG seed: 12345678
Forward Conv solutions available: 3
- id: 9 algo: 1, time: 16.1721 ms, ws: 0, name: ConvOclDirectFwdGen
- id: 85 algo: 1, time: 29.7198 ms, ws: 0, name: ConvDirectNaiveConvFwd
- id: 91 algo: 0, time: 507.536 ms, ws: 1048576, name: GemmFwdRest
Memory access fault by GPU node-1 (Agent handle: 0x1a25220) on address 0x7f1cd5820000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

This is Gemm. I suspect the reason is the same as discussed in #2808. Hopefully this will be fixed altogether with other #2808 failures. Do not close yet.

**Comment by huanrwan-amd** [comment]
Not able to reproduce with ROCm6.2.2:
```
MIOpenDriver convfp16 -n 4097 -c 2 -H 512 -W 512 -k 2 -y 8 -x 8 -p 0 -q 0 -u 8 -v 8 -l 1 -j 1 -m conv -g 1 -F 1 -t 1 -i 1 -S 91
PRNG seed: 12345678
Forward Conv solutions available: 2
- id: 91 algo: 0, time: 38.5802 ms, ws: 1048576, name: GemmFwdRest
- id: 85 algo: 1, time: 1000 ms, ws: 0, name: ConvDirectNaiveConvFwd
MIOpen Forward Conv. Algorithm: 0, Solution: 91/GemmFwdRest
GPU Kernel Time Forward Conv. Elapsed: 255.018555 ms (average)
stats: name, n, c, ho, wo, x, y, k, flopCnt, bytesRead, bytesWritten, GFLOPs, GB/s, timeMs
stats: fwd-conv8x8u8, 4097, 2, 64, 64, 8, 8, 2,  8592031744, 18446744069415633408, 67125248, 34, 72334911486, 255.018555
Forward Convolution Verifies OK on GPU reference (6.83101e-06 < 0.0082)
```
